sklearn/tests/test_discriminant_analysis.py

```diff
@@ -223,6 +225,60 @@ def test_lda_scaling():
                      'using covariance: %s' % solver)
 
 
+def test_lda_store_covariance():
+    # Test for slover 'lsqr' and 'eigen'
+    for solver in ('lsqr', 'eigen'):
+        clf = LinearDiscriminantAnalysis(solver=solver)
+        # The deprecated test will be asserted once the deprecation
+        # is complete.
+
+        # Test the actual attribute:
+        clf = LinearDiscriminantAnalysis(solver=solver,
+                                         store_covariance=True).fit(X6, y6)
+        assert_true(hasattr(clf, 'covariance_'))
+
+        assert_array_almost_equal(
+            clf.covariance_[0],
+            np.array([0.422222, 0.088889])
+        )
+
+        assert_array_almost_equal(
+            clf.covariance_[1],
+            np.array([0.088889, 0.533333])
+        )
+
+    # Test for slover svd, the default is to not set the covariances_ attribute
+    clf = LinearDiscriminantAnalysis(solver='svd').fit(X, y)
+    assert_false(hasattr(clf, 'covariance_'))
+
+    # Test the actual attribute:
+    clf = LinearDiscriminantAnalysis(solver=solver,
+                                     store_covariance=True).fit(X6, y6)
+    assert_true(hasattr(clf, 'covariance_'))
+
+    assert_array_almost_equal(
+        clf.covariance_[0],
+        np.array([0.422222, 0.088889])
+    )
+
+    assert_array_almost_equal(
+        clf.covariance_[1],
+        np.array([0.088889, 0.533333])
+    )
+
+
+def test_lda_deprecation():
+    for solver in ('lsqr', 'eigen'):
```

Comment:
Actually, I think this should be only svd. Because lsqr and eigen will not raise any error, they don't need `self.store_covariance`. and I was wondering if `testing.assert_raise_message` is more suitable here.